<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1500</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1500</width>
    <height>900</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>0</y>
      <width>661</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Diffie-Hellman</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="Alice_image">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>100</y>
      <width>141</width>
      <height>141</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Image/coding (1).png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="Bob_image">
    <property name="geometry">
     <rect>
      <x>1290</x>
      <y>100</y>
      <width>141</width>
      <height>141</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Image/coding.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="message_image">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>190</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Image/envelope.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="Alice_name">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>50</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Alice</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="Bob_name">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Bob</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>250</y>
      <width>160</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="AliceProperties">
     <item row="0" column="1">
      <widget class="QLineEdit" name="A_PKey_input"/>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="PKey_label">
       <property name="text">
        <string>Private key:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>1280</x>
      <y>250</y>
      <width>160</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="BobProperties">
     <item row="0" column="1">
      <widget class="QLineEdit" name="B_PKey_input"/>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="B_PKey_label">
       <property name="text">
        <string>Private key:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>1281</x>
      <y>320</y>
      <width>201</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Public keys</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>193</width>
       <height>61</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="p_label">
        <property name="text">
         <string>Prime numer p:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="G_label">
        <property name="text">
         <string>Generator number g:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="p_input"/>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="G_input"/>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="startButton">
    <property name="geometry">
     <rect>
      <x>1310</x>
      <y>450</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Start exchange key</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>310</y>
      <width>1231</width>
      <height>531</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="logLayout">
     <item>
      <widget class="QLabel" name="logLabel">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Log</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="Log">
       <property name="font">
        <font>
         <pointsize>20</pointsize>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="backButton">
    <property name="geometry">
     <rect>
      <x>1310</x>
      <y>490</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Back to main menu</string>
    </property>
   </widget>
   <zorder>message_image</zorder>
   <zorder>label</zorder>
   <zorder>Alice_image</zorder>
   <zorder>Bob_image</zorder>
   <zorder>Alice_name</zorder>
   <zorder>Bob_name</zorder>
   <zorder>gridLayoutWidget_2</zorder>
   <zorder>gridLayoutWidget_3</zorder>
   <zorder>groupBox</zorder>
   <zorder>startButton</zorder>
   <zorder>verticalLayoutWidget</zorder>
   <zorder>backButton</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1500</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
